scipy/integrate/_rules/_gauss_legendre.py

```diff
@@ -32,14 +38,21 @@ class GaussLegendreQuadrature(FixedRule):
      array([1.11022302e-16])
     """
 
-    def __init__(self, npoints):
+    def __init__(self, npoints, xp=None):
         if npoints < 2:
             raise ValueError(
                 "At least 2 nodes required for Gauss-Legendre cubature"
             )
 
         self.npoints = npoints
 
+        if xp is None:
+            xp = np_compat
+
+        self.xp = array_namespace(xp.empty(0))
+
     @cached_property
     def nodes_and_weights(self):
-        return roots_legendre(self.npoints)
+        # TODO: current converting to/from numpy
+        nodes, weights = roots_legendre(self.npoints)
+        return self.xp.asarray(nodes), self.xp.asarray(weights)
```

Comment:
is there test coverage for this? Typically, if we have to convert to NumPy and back, there are tests that need to be made cpu only.

Reply:
Ah so the "conversion to NumPy" is `self.npoints` giving us an `int`, right? So this is waiting for `roots_legendre` to get an `xp` parameter, or something like that? All good if so.